src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -627,8 +634,54 @@ public override BoundNode VisitBaseReference(BoundBaseReference node)
                 : FramePointer(node.Syntax, _topLevelMethod.ContainingType); // technically, not the correct static type
         }
 
+        private void RemapLocalFunction(CSharpSyntaxNode syntax, MethodSymbol symbol, out BoundExpression receiver, out MethodSymbol method)
+        {
+            Debug.Assert(symbol.MethodKind == MethodKind.LocalFunction);
+
+            var constructed = symbol as SubstitutedMethodSymbol;
+            if (constructed != null)
+            {
+                RemapLocalFunction(syntax, constructed.ConstructedFrom, out receiver, out method);
+                return;
+            }
+
+            method = _localFunctionMap[(LocalFunctionSymbol)symbol];
+            var translatedLambdaContainer = method.ContainingType;
+            var containerAsFrame = translatedLambdaContainer as LambdaFrame;
+
+            // Rewrite the lambda expression (and the enclosing anonymous method conversion) as a delegate creation expression
+            NamedTypeSymbol constructedFrame = (object)containerAsFrame != null ?
+                translatedLambdaContainer.ConstructIfGeneric(StaticCast<TypeSymbol>.From(_currentTypeParameters)) :
+                translatedLambdaContainer;
+
+            // for instance lambdas, receiver is the frame
+            // for static lambdas, get the singleton receiver
+            if (containerAsFrame?.SingletonCache == null)
```

Comment:
Don't you need to check the kind of this particular lambda/function, rather than whether *anything* on the frame is static?